tests/src/Interop/PInvoke/Structures/DelegatePInvoke/MarshalSeqStruct/SeqStructDelRevPInvokeNative.h

```diff
@@ -0,0 +1,884 @@
+//typedef float FLOAT;
+//typedef double DOUBLE;
+#endif
+
+struct INNER2 // size = 12 bytes
+{
+    INT f1;
+    FLOAT f2;
+    LPCSTR f3;
+};
+
+void ChangeINNER2(INNER2* p)
+{
+    p->f1 = 77;
+    p->f2 = 77.0;
+    char* temp = "changed string";
+    size_t len = strlen(temp);
+    LPCSTR str = (LPCSTR)CoTaskMemAlloc( sizeof(char)*(len+1) );
+    strcpy_s((char*)str,len+1,temp);
+    p->f3 = str;
+}
+
+void PrintINNER2(INNER2* p, char* name)
+{
+    printf("\t%s.f1 = %d\n", name, p->f1);
+    printf("\t%s.f2 = %f\n", name, p->f2);
+    printf("\t%s.f3 = %s\n", name, p->f3);
+}
+
+bool IsCorrectINNER2(INNER2* p)
+{
+    if(p->f1 != 1)
+        return false;
+    if(p->f2 != 1.0)
+        return false;
+    if(memcmp(p->f3, "some string",11*sizeof(char)) != 0 )
+        return false;
+    return true;
+}
+
+struct InnerExplicit 
+{
+#ifdef WINDOWS
+    union
+    {
+        INT f1;
+        FLOAT f2;
+    };
+    CHAR _unused0[4];
+    LPCSTR f3;
+#else
+    union
+    {
+        INT f1;
+        FLOAT f2;
+    };
+    INT _unused0;
+    LPCSTR f3;
+#endif 
+};
+
+void PrintInnerExplicit(InnerExplicit* p, char* name)
+{
+    printf("\t%s.f1 = %d\n", name, p->f1);
+    printf("\t%s.f2 = %f\n", name, p->f2);
+    printf("\t%s.f3 = %s\n", name, p->f3);
+}
+
+void ChangeInnerExplicit(InnerExplicit* p)
+{
+    p->f1 = 77;
+
+    char* temp = "changed string";
+    size_t len = strlen(temp);
+    LPCSTR str = (LPCSTR)CoTaskMemAlloc( sizeof(char)*(len+1) );
+    strcpy_s((char*)str,len+1,temp);
+    p->f3 = str;
+}
+
+struct InnerArraySequential
+{
+    InnerSequential arr[NumArrElements];
+};
+
+void PrintInnerArraySequential(InnerArraySequential* p, char* name)
+{
+    for(int i = 0; i < NumArrElements; i++)
+    {
+        printf("\t%s.arr[%d].f1 = %d\n", name, i, (p->arr)[i].f1);
+        printf("\t%s.arr[%d].f2 = %f\n", name, i, (p->arr)[i].f2);
+        printf("\t%s.arr[%d].f2 = %s\n", name, i, (p->arr)[i].f3);
+    }
+}
+
+void ChangeInnerArraySequential(InnerArraySequential* p)
+{
+    char* lpstr = "changed string";
+    LPSTR temp;
+    for(int i = 0; i < NumArrElements; i++)
+    {
+        (p->arr)[i].f1 = 77;
+        (p->arr)[i].f2 = 77.0;
+
+        size_t size = sizeof(char) * (strlen(lpstr) + 1);
+        temp = (LPSTR)CoTaskMemAlloc( size );
+        memset(temp, 0, size);
+        if(temp)
+        {
+            strcpy_s((char*)temp,size,lpstr);
+            (p->arr)[i].f3 = temp;
+        }
+        else
+        {
+            printf("Memory Allocated Failed!");
+        }
+    }
+}
+
+bool IsCorrectInnerArraySequential(InnerArraySequential* p)
+{
+    for(int i = 0; i < NumArrElements; i++)
+    {
+        if( (p->arr)[i].f1 != 1 )
+            return false;
+        if( (p->arr)[i].f2 != 1.0 )
+            return false;
+    }
+    return true;
+}
+
+#ifdef WINDOWS
+#ifdef _WIN64
+union OUTER3 // size = 32 bytes
+{
+    struct InnerSequential arr[2];
+    struct
+    {
+        CHAR _unused0[24];
+        LPCSTR f4;
+    };
```

Comment:
Warning about unnamed struct here.
`warning C4201: nonstandard extension used: nameless struct/union`
I tried to add Inner to this struct, issue would happen, f4 will be unreachable by code example `p->f4`.
This will cause issue in code. As if the struct is named, we cannot access f4 by this way.

Reply:
Is it ok to just define OUTER3 as following but not using `union` in X64:
`struct OUTER3 // size = 28 bytes
{
    struct InnerSequential arr[2];
    LPCSTR f4;
};`

Reply:
I don't understand the issue then. Why can't we fix access of that field in code?

Reply:
@AaronRobinsonMSFT The definition for OUTER3 is different in X64 & X86 build according to the code.
If we added Inner to the unnamed struct in X64, then the access to the f4 field will be different in X86 & X64.
X64: `p.Inner.f4`
X86:`p.f4`

If I changed the X86 struct OUTER similiar to X64, added Inner struct, then move f4 into Inner struct. 
`struct OUTER3 // size = 28 bytes
{
    struct InnerSequential arr[2];
    struct Inner
    {
        LPCSTR f4;
    }
};`
Build passed, but the C# test failed with BadImageFormatException. 
And I'm not sure why OUTER3 need to be different in X86 & X64.
So I'm not sure how to fix the C# test.